### ISSUE TITLE
support for SFTP

### ISSUE THREAD

**Issue by john30** [JCMais/node-libcurl#136]
when using node 10, it seems support for sftp protocol is there on Windows but not on Linux

**Comment by JCMais** [comment]
can you try again that? If the error persists, can you share a code that reproduces this?


Thanks

**Comment by john30** [comment]
here is some code to reproduce:
```
var libcurl = require("node-libcurl");
var url = 'sftp://test.rebex.net/';
var username = 'demo';
var password = 'password';
var handle = new libcurl.Easy();
handle.setOpt(libcurl.Curl.option.USERNAME, username);
handle.setOpt(libcurl.Curl.option.PASSWORD, password);
handle.setOpt(libcurl.Curl.option.FAILONERROR, true);
handle.setOpt(libcurl.Curl.option.URL, url);
var found = 0;
handle.setOpt(libcurl.Curl.option.WRITEFUNCTION, function (buff, size, nmemb) {
    found++;
    return size * nmemb;
});
var code = handle.perform();
handle.close();
if (code !== libcurl.Curl.code.CURLE_OK) {
    var err = libcurl.Easy.strError(code);
    console.log('error: ', code, err);
} else {
    console.log('OK', found);
}
```

this gives the following results:

- Linux, node v9.3.0:
error:  1 Unsupported protocol
- Linux, node v10.4.1:
error:  1 Unsupported protocol
- Linux, node v10.11.0:
error:  1 Unsupported protocol
- Windows, node v8.11.3:
OK 5
- Windows, node v10.11.0:
Segmentation fault

**Comment by JCMais** [comment]
Thanks for the code, I will do some digging later this week .

**Comment by imade** [comment]
I had same issue with macOS High Sierra (10.13.6) and node v10.15.0 but with the steps below I got it working.

Tested node-libcurl SFTP based on the [official example code](https://github.com/JCMais/node-libcurl/blob/develop/examples/ssh.js)

Output:
```
sftp-test $ node index.js 
* Rebuilt URL to: sftp://<hidden>:<hidden>@<hidden>/
* Protocol "sftp" not supported or disabled in libcurl
* Closing connection -1
1 false 'Unsupported protocol'
```

Next verified that my local curl itself actually supports SFTP:
```
$ curl -k "sftp://<hidden>:22/myuser/somefolder/" --user "<hidden>:<hidden>"
```
This gave similar error: `Protocol sftp not supported or disabled in libcurl`

Based on [this blog post](http://andrewberls.com/blog/post/adding-sftp-support-to-curl) I found out how to enable sftp with curl. I didn't follow the steps exactly as I got it working using brew.

First verified that my machine has `libssh2` available when I want to install curl.
```
$ brew info curl
```

This should show green check mark next to `libssh2`.

Next installed curl itself:
```
$ brew install --with-libssh2 curl
$ echo 'export PATH="/usr/local/opt/curl/bin:$PATH"' >> ~/.bash_profile
$ source ~/.bash_profile
$ curl -k "sftp://<hidden>:22/myuser/somefolder/" --user "<hidden>:<hidden>"
```

As final step I re-installed my node-libcurl dependency an ensured it would link to new system libcurl.

```
$ export LDFLAGS="-L/usr/local/opt/curl/lib"
$ export CPPFLAGS="-I/usr/local/opt/curl/include"
$ npm i
```

Also worth noting is that my package.json has a preinstall step for node-libcurl:
```
"scripts": {
  "preinstall": "npm install --build-from-source node-libcurl"
}
```

After all these steps the original test code started working properly.

Given macOS and Linux are very similar it might help with the Linux issues too.
